### DISCUSSION TITLE
Possible to use Profisafe alongside RTDE

### DISCUSSION THREAD

**Discussion by LukGen** [UniversalRobots/Universal_Robots_ROS2_Driver#473] [Q&A]
Hello,

Is it possible to use the ur ros2 driver alongside with profinet. For instance if you want to control the movement of your robot with ros2 but still want to adress safety-issues with a classic plc?

**Comment by destogl** [comment]
I don't understand how this should be "connected" at all? Can you draw a small system architecture?

**Reply by gavanderhoorn** [reply]
@destogl: the RTDE infrastructure is used by various fieldbus implementations on UR controllers.

This is the cause of EtherNet/IP not working correctly when using the driver fi (and the other way around: the driver not working when EtherNet/IP is active/configured).

The question posted by the OP makes sense to me in that context.

**Reply by LukGen** [reply]
Profinet builds on tcp/ip and can therefore be transmitted via ethernet. The system architecture is therefore rather simple. The UR-Controller, a ipc and the plc would be connected to a switch via ethernet and that's it.

**Reply by fmauch** [reply]
related: https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/issues/56#issuecomment-566029214 and apparently douple post with https://forum.universal-robots.com/t/profisafe-alongside-ros2-driver-or-rtde/23750
